pkg/extender/extender.go

```diff
@@ -241,8 +243,10 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
 				for _, volumeNode := range volumeInfo.DataNodes {
 					for _, driverNode := range driverNodes {
 						if volumeNode == driverNode.StorageID {
-							// preferRemoteNodeOnly applies only to volumes with NeedsAntiHyperconvergence
-							if e.volumePrefersRemoteOnly(volumeInfo) && volumeInfo.NeedsAntiHyperconvergence {
+							// prefersRemoteNode and preferRemoteNodeOnly apply only to volumes with NeedsAntiHyperconvergence
+							if volumeInfo.NeedsAntiHyperconvergence &&
+								e.volumePrefersRemoteNode(volumeInfo) &&
```

Comment:
Is it "PrefersRemoteNode OR PrefersRemoteNodeOnly" or all 3 conditions need to match?

Reply:
Yes Dipti all 3 conditions needs to match to strictly enforce anti-hyperconvergence.
volumeInfo.NeedsAntiHyperconvergence -> tells us if volume is sharedV4 service volume
PrefersRemoteNode -> new SC parameter. If it's value is false, ignore the parameter preferRemoteNodeOnly
PrefersRemoteNodeOnly -> if true, strictly enforce anti-hyperconvergence i.e. filter out nodes which have replicas of the volume.